### PR TITLE
GitHub Pages provider - Add option 'github-url'

### LINKED ISSUE #627: Support GitHub Enterprise
I am using a hosted Github enterprise version.

Unfortunately, it seems that `dpl` has github.com hardcoded somewhere

Examples:

```
dpl --provider=pages \
      --local-dir=site \
      --github-token=MY_TOKEN \
      --skip_cleanup \
      --repo=ORG_NAME/REPO_NAME.git 

...

Password for 'https://MY_TOKEN@github.com':
```

```
dpl --provider=pages \
      --local-dir=site \
      --github-token=MY_TOKEN \
      --skip_cleanup \
      --repo=https://github.myorg.io/ORG_NAME/REPO_NAME.git 

...

Couldn't push the build to github.com/https://github.myorg.io/ORG_NAME/REPO_NAME.git.git:gh-pages

```

### PR DESCRIPTION
**Pull request by xakraz**
Fix #627

### CODE REVIEW

**Comment by xakraz** [comment]
Tested "Manually" on our hosted github enterprise.

File: lib/dpl/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module DPL
-  VERSION = '1.8.32'
+  VERSION = '1.8.33'
```
Comment (BanzaiMan) [inline lib/dpl/version.rb]: Please remove this. We manage the releases, and we cannot merge this PR with this bit.

**Comment by xakraz** [comment]
@BanzaiMan I have submitted a PR for the Doc as asked: https://github.com/travis-ci/docs-travis-ci-com/pull/1216

**Comment by xakraz** [comment]
Renamed the branch
New PR is https://github.com/travis-ci/dpl/pull/641

Sorry for the inconvenience ...

**Pull request closed without merging**